intercepting-filter/src/main/java/com/iluwatar/intercepting/filter/Client.java

```diff
@@ -37,20 +35,20 @@
 import javax.swing.SwingUtilities;
 
 /**
- * The Client class is responsible for handling the input and running them through filters inside
- * the {@link FilterManager}.
+ * The Client class is responsible for handling the input and running them through filters inside the
+ * {@link FilterManager}.
  *
- * This is where {@link Filter}s come to play as the client pre-processes the request before being
- * displayed in the {@link Target}.
+ * This is where {@link Filter}s come to play as the client pre-processes the request before being displayed in the
+ * {@link Target}.
  * 
  * @author joshzambales
  *
  */
-public class Client extends JFrame {
+public class Client extends JFrame { // NOSONAR
```

Comment:
Sonarqube reported "This class has 6 parents which is greater than 5 authorized"
Here is the inheritance hierarchy. 
FileSelectorJFrame -> JFrame -> Frame -> Window -> Container -> Component implements ImageObserver, MenuContainer, Serializable

I believe, it is most common to subclass JFrame for creating window. 
Alternate design by following "Use composition over inheritance". I thought it would make more changes and may spoil the brevity of explaining the design pattern. 
Please let me know if you want me to think about the alternate design